src/SelfRoleManager.ts

```diff
@@ -200,8 +204,10 @@ export class SelfRoleManager extends EventEmitter {
       }
     }
 
+    const listener = this.#rolesChangesListener(channelOptions.rolesToEmojis);
+    this.client.on(Events.GuildMemberUpdate, listener );
     if (!this.channels.has(channel.id)) {
-      this.channels.set(channel.id, { ...channelOptions, message: { ...channelOptions.message, id: message.id } });
+      this.channels.set(channel.id, { ...channelOptions, message: { ...channelOptions.message, id: message.id }, _rolesChangesListener: listener });
```

Comment:
The listener is an internal process and should not be in the facing API. I would not add it to the `channels` collection. I'd rather create a private readonly collection for internal stuff like this. Are you up to this task? (sounds like a quest given by a PNG tbh but well, gotta rp some times)

Reply:
I agree to hide the listener from the public API. But instead of creating a new collection for internal stuff, why not reuse the collection `channel` with an attribute `options` for the channel options (all existing options) and add other attributes for internal purposes.
This way, we don't need to manage 2 collections at register and unregister time.

Reply:
That would be a good idea. You would need to change a few things here and there though. This requires thorough checks (and this library is not unit tested iirc).

Reply:
The change was easy, the collection `channels` is called only twice and one required options.

To minimize changes, I access directly to `ChannelOptions` with the `?.options` syntax. 

```ts
const channelOptions = this.channels.get(userAction.message.channelId)?.options;
if (!channelOptions) return;
```
I can use this as well, but I like it less.
```ts
const { options: channelOptions } = this.channels.get(userAction.message.channelId);
if (!channelOptions) return;
```